src/dune/coq_rules.ml

```diff
@@ -362,12 +392,38 @@ let install_rules ~sctx ~dir s =
                  vofile) ))
     |> List.rev_append coq_plugins_install_rules
 
-let coqpp_rules ~sctx ~build_dir ~dir (s : Dune_file.Coqpp.t) =
-  let cc = create_ccoq sctx ~dir in
+let coqpp_rules ~sctx ~build_dir ~dir (s : Coqpp.t) =
+  let coqpp = resolve_program sctx ~dir ~loc:s.loc "coqpp" in
   let mlg_rule m =
     let source = Path.build (Path.Build.relative dir (m ^ ".mlg")) in
     let target = Path.Build.relative dir (m ^ ".ml") in
     let args = [ Command.Args.Dep source; Hidden_targets [ target ] ] in
-    Command.run ~dir:(Path.build build_dir) cc.coqpp args
+    Command.run ~dir:(Path.build build_dir) coqpp args
   in
   List.map ~f:mlg_rule s.modules
+
+let extract_rules ~sctx ~build_dir ~dir ~dir_contents (s : Extract.t) =
+  let cctx =
+    let wrapper_name = "DuneExtraction" in
+    let theories_deps =
+      let scope = SC.find_scope_by_dir sctx dir in
+      let coq_lib_db = Scope.coq_libs scope in
+      Coq_lib.DB.requires_for_user_written coq_lib_db s.buildable.theories
+    in
+    Context.create sctx ~dir ~wrapper_name ~theories_deps s.buildable
+  in
+  let coq_module =
+    let coq = Dir_contents.coq dir_contents in
+    Coq_sources.extract coq s
+  in
+  let file_flags = Context.coqc_file_flags cctx in
+  let ml_targets =
+    Extract.ml_target_fnames s |> List.map ~f:(Path.Build.relative build_dir)
```

Comment:
build_dir is the root context dir, used in run, this should be `dir` I believe.

Reply:
Yup fixed.